### ISSUE TITLE
Save the trained neural networks and the pruning masks

### ISSUE THREAD

**Issue by lecode-official** [lecode-official/pytorch-lottery-ticket-hypothesis#1]
Right now, the models are trained and are not stored to disk. The models, as well as the pruning masks should be stored after each iteration of the iterative magnitude pruning algorithm.
